### PR TITLE
bob: add completion

### PR DESCRIPTION
**Pull request by sudosubin**
## Description of changes

Add `bash`, `zsh` shell completion for `bob`.
https://bob.build/docs/next/getting-started/installation#autocompletion

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]
Nit: that postInstall snippet should probably not be done when cross-compiling (because that'll fail).

File: pkgs/development/tools/build-managers/bob/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```
Comment (wegank) [inline pkgs/development/tools/build-managers/bob/default.nix:1]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles }:
```

File: pkgs/development/tools/build-managers/bob/default.nix [hunk 2]
```diff
@@ -17,6 +17,14 @@ buildGoModule rec {
 
   excludedPackages = [ "example/server-db" "test/e2e" "tui-example" ];
 
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
```
Comment (wegank) [inline pkgs/development/tools/build-managers/bob/default.nix:22]:
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

Comment (sudosubin) [reply pkgs/development/tools/build-managers/bob/default.nix]: thansk for the suggestion. I didn't know how to safely wrap the script for cross-compile.

**Review by bjornfor — APPROVED** [review]

**Pull request merged**